modules/standard/IO.chpl

```diff
@@ -7018,6 +7043,105 @@ iter fileReader.lines(stripNewline = false) {
   this.unlock();
 }
 
+@chpldoc.nodoc
+iter fileReader.lines(
+  param tag: iterKind,
+  stripNewline = false,
+  targetLocales: [?tld] locale = [this._home,]
+): string
+  where tag == iterKind.standalone
+{
+  on this._home {
+    try! this.lock();
+
+    var f = chpl_fileFromReaderOrWriter(this);
+
+    const myStart = qio_channel_get_start_pos(this._channel_internal),
+          myEnd = qio_channel_get_end_pos(this._channel_internal),
+          myBounds = myStart..<(try! min(myEnd, f.size));
+
+    if targetLocales.size == 1 && targetLocales.first == this._home {
+      // single locale case on this fileReader's home locale
+      const nTasks = here.maxTaskPar,
```

Comment:
IMO this should compute nTasks using `dataParTasksPerLocale`

See for example:

https://github.com/chapel-lang/chapel/blob/2259802456cd8ed2c7e3795d48ecaa217c44327c/modules/standard/Sort.chpl#L1786-L1800

Also, IMO we pretty badly need a helper to do this that is documented; see #12784.